articles/connections/apple-siwa/set-up-apple.md

```diff
@@ -0,0 +1,124 @@
+
+2. Choose **Services IDs** and click **Continue**. Fill in the description and identifier (`com.customdomain.webapp`).
+3. After checking **Sign In with Apple**, click on **Configure** and define your **Web Domain** (`customdomain.com`) and your **Return URL**. Make sure that your Return URL follows this format: `https://customdomain.com/login/callback`.
+
+    ![Configure URLs](/media/articles/connections/social/apple/apple-configureurls.jpg)
+
+4. Click **Save**, **Continue**, and then click **Register**.
+
+### Verify domain with Apple
+
+1. On the **Certificates, IDs, & Profiles** page, click your newly created Services ID.
+2. Click the **Configure** button next to the **Sign In with Apple** feature.
+
+3. On the **Redirect URI/Domain Validation** page click **Download**.
+4. Copy or move the `apple-developer-domain-association.txt` file to your application server and make it accessible at `https://customdomain.com/.well-known/apple-developer-domain-association.txt`.
+1. When this is ready, click **Verify** on your Services ID configuration page. If no error message is shown, your domain has been successfully verified.
+
+## Set up Signing Key
+
+1. Go to **Keys** under the **Certificates, Identifiers, & Profiles** section of your Apple developer dashboard.
+2. Click on the **blue plus icon** to add a new key.
+3. Enter a **Key Name** and check the **Sign In with Apple** option.
+4. Click **Configure** to make sure the **Choose a Primary App ID** field is filled with the correct App ID.
+5. Click **Save**, **Continue**, and then **Register**.
+6. On the page to which you're redirected after registering, make a note of the Key ID. Then, download the key, move it to its own directory, and rename the key file that you downloaded to `authkey.p8`. 
+7. Use a JWT library with support for the ES256 algorithm to generate the client secret from the key. The example below uses `Node.js`, but libraries for other languages can be used in a similar fashion. To use this script, create a new file called `generate-secret.js` in the same directory as the downloaded key and add the following code:
+
+    ```js
+    const jwt = require("jsonwebtoken");
+    const fs = require("fs");
+
+    const privateKey = fs.readFileSync("./authkey.p8");
+    const token = jwt.sign({}, privateKey, {
+    algorithm: "ES256",
+    expiresIn: "60 days",
+    audience: "https://appleid.apple.com",
+    issuer: "TEAM_ID",
+    subject: "com.customdomain.webapp",
+    keyid: "KEY_ID"
+    });
+
+    console.log("The token is:", token);
+    ```
+
+    Replace the following values:
+
+    * `com.customdomain.webapp` with the identifier for your Services ID
+    * `TEAM_ID` with your Team ID
+    * `KEY_ID` with your Key ID
+
+8. Once your script is ready, run the following on the command line to generate your client secret:
+
+    `node generate-secret.js`
+
+    This gives you the last of the items you need to provide when setting up the connection in Auth0's dashboard.
```

Comment:
I am confused, the `app_secret` that we use to generate the `client_secret` sent to apple on the exchange:
https://github.com/auth0/auth0-server/blob/4c8b6ba8e9d8bb43553586249e7b09a98e709072/lib/auth/strategies/apple.js#L37

Is that really the token resulted from running that script? How is the ES alg verification working with that on their end? 🤔 

cc: @josh-cain @panva

Reply:
`## Set up Signing Key`

This section should only make the developer download the private key from apple, copy its ASCII content to clibpoard and paste it in the manage dashboard. No JWT signing, thats all handled by the connection in auth0

Reply:
It's not, had it removed, but we got our PR's crossed.  Apologies for the confusion.